include/nbl/asset/utils/CGLSLCompiler.h

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_ASSET_C_GLSL_COMPILER_H_INCLUDED_
+#define _NBL_ASSET_C_GLSL_COMPILER_H_INCLUDED_
+
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl/asset/utils/IShaderCompiler.h"
+
+namespace nbl::asset
+{
+
+class NBL_API2 CGLSLCompiler final : public IShaderCompiler
+{
+	public:
+		IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_GLSL; };
+
+		CGLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system);
+
+		struct SOptions : IShaderCompiler::SOptions
+		{
+			void setCommonData(const IShaderCompiler::SOptions& opt)
+			{
+				static_cast<IShaderCompiler::SOptions&>(*this) = opt;
+			}
+
+			virtual IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_GLSL; };
```

Comment:
the final (which you can add to this struct btw) derivation doesn't need `virtual`

Reply:
yea, I leaked an old habbit of mine into the code.
my logic was even if it's a derivation it's a virtual function.
will fix it